File: mvvm/src/main/java/com/lookballs/mvvm/core/BaseAdapter.java
package com.lookballs.mvvm.core;

import android.view.View;

import androidx.annotation.NonNull;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.module.LoadMoreModule;
import com.chad.library.adapter.base.viewholder.BaseViewHolder;

public class BaseAdapter<AB> extends BaseQuickAdapter<AB, BaseAdapter.BaseHolder> implements LoadMoreModule {
    public BaseAdapter(int layoutResId) {
        super(layoutResId);
    }

    @Override
    protected void convert(@NonNull BaseHolder holder, AB ab) {

    }

    public static class BaseHolder extends BaseViewHolder {
        public BaseHolder(@NonNull View view) {
            super(view);
        }
    }
}

File: mvvm-kt/src/main/java/com/lookballs/mvvm/core/BaseAdapter.kt
package com.lookballs.mvvm.core

import android.view.View
import com.chad.library.adapter.base.BaseQuickAdapter
import com.chad.library.adapter.base.module.LoadMoreModule
import com.chad.library.adapter.base.viewholder.BaseViewHolder

class BaseAdapter<AB>(layoutResId: Int) : BaseQuickAdapter<AB, BaseAdapter.BaseHolder>(layoutResId), LoadMoreModule {
    override fun convert(holder: BaseHolder, ab: AB) {

    }

    class BaseHolder(view: View) : BaseViewHolder(view)
}